tests/builder/test_base.py

```diff
@@ -122,11 +122,7 @@ def test_run_exception(builder):
     if sys.version_info >= (3, 6):
         builder.build.assert_called()
     # bulid should be called 3 times for retry setting.
-    calls = [
-        mock.call({'name': 'a'}),
-        mock.call({'name': 'a'}),
-        mock.call({'name': 'a'}),
-    ]
+    calls = [mock.call({'name': 'a'}, retry=3)] * 3
```

Comment:
`assert_has_calls` does a *subset* of the calls – which means if there any extra calls, the assert still passes.

I fixed that by also asserting on `Mock.call_count`.